core/cloudflow-akka-testkit/src/main/scala/cloudflow/akkastream/testkit/TestContext.scala

```diff
@@ -50,10 +50,15 @@ private[testkit] case class TestContext(
     StreamletDefinition("appId", "appVersion", streamletRef, "streamletClass", List(), volumeMounts, config)
 
   @deprecated("Use `sourceWithCommittableContext` instead.", "1.3.4")
-  override def sourceWithOffsetContext[T](inlet: CodecInlet[T]): cloudflow.akkastream.scaladsl.SourceWithOffsetContext[T] =
+  override def sourceWithOffsetContext[T](
```

Comment:
If we just change the name and create a brand new API method we'lll avoid a breaking change. What do you think?

Reply:
We can, but I was hoping not to do this by introducing default values

Reply:
If we need to break things it seems it makes more sense to change `Codec`. The `DataConverter` introduces a too similar thing to codec, let's discuss.

Reply:
```scala
val inlet = ProtoInlet(...).handleErrors { bytes =>
  // do what you like here, return unit
}
```

Reply:
The reason I was doing it this way is to allow users to overwrite default implementation, which is harder if we will try to build it in codec. It is not about extending codec capabilities, but rather allowing users to customize behavior. We can not really foresee all of possible things that the users will want to do

Reply:
it's the same right, just add a function that handles it to the codec, with some convenience. Also, there is not much you can choose really. all you can do is `Array[Byte] => Unit`.

Reply:
And I think it makes sense to (at least)  add a `BytesInlet`, which is like `Codec[Array[Byte]]`. Then you can read in any way you like, but that kind of inlet has no schema so will never match an outlet, for safety.

Reply:
I was suggesting this to you several times, but you were always hesitant about it. There are several things that are possible here. Also you are correct, the only thing we can do is  Array[Byte] => Unit, but there are a couple of things here:
1. How do you get to this point - just syntactic validation or both syntactic and semantic.
2. Which side effect do you want - just logging, writing to Dead-letter topic, etc.

Reply:
Also, the approach that I have taken provides more fine grained configuration. I can control error processing on per stream basis, rather then doing it globally (per streamlet or even pipeline)